cmake/Check128BitCas.cmake

```diff
@@ -25,18 +25,16 @@
 
 set(CHECK_PROGRAM
     "
-    int
-    main()
+    int main(void)
     {
         __int128_t x = 0;
-        __sync_bool_compare_and_swap(&x,0,10);
-
-        return 0;
+        return __sync_bool_compare_and_swap(&x,0,10);
     }
     "
 )
 
 include(CheckCSourceCompiles)
+set(CMAKE_REQUIRED_FLAGS "-march=native")
 check_c_source_compiles("${CHECK_PROGRAM}" TS_HAS_128BIT_CAS)
 
```

Comment:
It might be wise to `unset(CMAKE_REQUIRED_FLAGS)`, because otherwise the variable persists and could influence the next call to `check_c_source_compiles` unintentionally.